debian/apport-general-hook.py

```diff
@@ -0,0 +1,51 @@
+import json
+import logging
+
+
+def get_ec2_data(instance_ci_data):
+    """
+    provided cloud-init data from /run/cloud/instance-data.json
+    populate a smaller set of information helpful for debugging
+    """
+    stripped_data = {}
+    instance_identity_keys = (
+        "architecture",
+        "billingProducts",
+        "imageId",
+        "region",
+    )
+    metadata_keys = ("instance-type",)
+    for key in instance_identity_keys:
+        stripped_data[key] = instance_ci_data["ds"]["dynamic"][
+            "instance-identity"
+        ]["document"][key]
+    for key in metadata_keys:
+        stripped_data[key.capitalize()] = instance_ci_data["ds"]["meta-data"][
+            key
+        ]
+    stripped_data["region"] = instance_ci_data["v1"]["region"]
+    return stripped_data
+
+
+def add_info(report, ui):
+    """
+    add a subset of non-sensitive cloud-init data
+    """
+    cloud = ""
+    retrieved_data = {}
+    try:
+        with open("/run/cloud-init/instance-data.json", "r") as fopen:
+            full_data = json.load(fopen)
+    except FileNotFoundError:
+        logging.getLogger().warning(
+            "cloud-init run data not found on system. "
+            "Unable to add cloud-specific data."
+        )
+        return
+    # there are a lot of possibilities here, per cloud.
+    # deferring to per_cloud functions based on identification
```

Comment:
This comment doesn't line up with the code. Are more functions planned in this pull request?

Reply:
Your comment triggered me to nearly completely rewrite this hook. Now it will add `CloudName`, `CloudPlatform`, `CloudRegion` for all clouds.